src/main/java/io/vertx/ext/amqp/AmqpClientOptions.java

```diff
@@ -0,0 +1,559 @@
+   */
+  @Override
+  public AmqpClientOptions setReusePort(boolean reusePort) {
+    super.setReusePort(reusePort);
+    return this;
+  }
+
+  /**
+   * @see ProtonClientOptions#setTcpCork(boolean)
+   */
+  @Override
+  public AmqpClientOptions setTcpCork(boolean tcpCork) {
+    super.setTcpCork(tcpCork);
+    return this;
+  }
+
+  /**
+   * @see ProtonClientOptions#setTcpFastOpen(boolean)
+   */
+  @Override
+  public AmqpClientOptions setTcpFastOpen(boolean tcpFastOpen) {
+    super.setTcpFastOpen(tcpFastOpen);
+    return this;
+  }
+
+  /**
+   * @see ProtonClientOptions#setTcpQuickAck(boolean)
+   */
+  @Override
+  public AmqpClientOptions setTcpQuickAck(boolean tcpQuickAck) {
+    super.setTcpQuickAck(tcpQuickAck);
+    return this;
+  }
+
+  /**
+   * @see ProtonClientOptions#removeEnabledSecureTransportProtocol(String)
+   */
+  @Override
+  public AmqpClientOptions removeEnabledSecureTransportProtocol(String protocol) {
+    super.removeEnabledSecureTransportProtocol(protocol);
+    return this;
+  }
+
+  /**
+   * @see ProtonClientOptions#setEnabledSecureTransportProtocols(Set)
+   */
+  @Override
+  public AmqpClientOptions setEnabledSecureTransportProtocols(Set<String> enabledSecureTransportProtocols) {
+    super.setEnabledSecureTransportProtocols(enabledSecureTransportProtocols);
+    return this;
+  }
+
+  /**
+   * @see ProtonClientOptions#setVirtualHost(String)
+   */
+  @Override
+  public AmqpClientOptions setVirtualHost(String virtualHost) {
+    super.setVirtualHost(virtualHost);
+    return this;
+  }
+
+  /**
+   * @see ProtonClientOptions#setSniServerName(String)
+   */
+  @Override
+  public AmqpClientOptions setSniServerName(String sniServerName) {
+    super.setSniServerName(sniServerName);
+    return this;
+  }
+
+  /**
+   * @see ProtonClientOptions#setHeartbeat(int)
+   */
+  @Override
+  public AmqpClientOptions setHeartbeat(int heartbeat) {
+    super.setHeartbeat(heartbeat);
+    return this;
+  }
+
+  /**
+   * @see ProtonClientOptions#setMaxFrameSize(int)
+   */
+  @Override
+  public AmqpClientOptions setMaxFrameSize(int maxFrameSize) {
+    super.setMaxFrameSize(maxFrameSize);
+    return this;
+  }
+
+  private String getFromSysOrEnv(String key) {
+    String sys = System.getProperty(key);
+    if (sys == null) {
+      return System.getenv(key.toUpperCase().replace("-", "_"));
+    }
+    return sys;
+  }
+
+  private int getPortFromSysOrEnv() {
+    String s = getFromSysOrEnv("amqp-client-port");
+    if (s == null) {
+      return 5672;
+    } else {
+      return Integer.parseInt(s);
+    }
+  }
+
+  /**
+   * Sets the max buffered messages. This message can be used to configure the initial credit of a receiver.
+   *
+   * @param maxBufferSize the max buffered size, must be positive. If not set, default credit is used.
+   * @return the current {@link AmqpClientOptions} instance.
+   */
+  public AmqpClientOptions setMaxBufferedMessages(int maxBufferSize) {
```

Comment:
This feels more like a receiver option.

Reply:
Agreed. It's a left-over from when I didn't have receiver options.